var/spack/repos/builtin/packages/netlib-scalapack/package.py

```diff
@@ -21,12 +23,43 @@ class NetlibScalapack(Package):
     depends_on('cmake')
     depends_on('mpi')
     depends_on('lapack')
+    depends_on('blas')
 
     def install(self, spec, prefix):
+        # Look for both shared and static libraries, since we don't know what was built
+        exts = ['.a', '.so']
+        if '+shared' in spec:
+            exts.reverse()
+
+        # Find BLAS for the not-so-powerful CMake script
```

Comment:
same as in another PR, i would use features from #657 namely `self.spec.blas_shared_lib` and alike.